postgrest/base_request_builder.py

```diff
@@ -155,6 +156,69 @@ def from_http_request_response(
         count = cls._get_count_from_http_request_response(request_response)
         return cls(data=data, count=count)
 
+    @classmethod
+    def from_dict(cls: Type[APIResponse], dict: Dict[str, Any]) -> APIResponse:
+        keys = dict.keys()
+        assert len(keys) == 3 and "data" in keys and "count" in keys and "error" in keys
+        return cls(
+            data=dict.get("data"), count=dict.get("count"), error=dict.get("error")
+        )
+
+
+class SingleAPIResponse(BaseModel):
```

Comment:
Are there any type errors if you inherit from APIResponse here? We can avoid the all the method duplication then right? We would then have to just change the types and override the `data` validator. What do you think?